File: lib/src/presentation/auth/widget/sign_up/sign_in_button.dart
import 'package:flutter/material.dart';
import 'package:silver_heart/src/presentation/auth/view/sign_in.dart';

class SignInButton extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(top: 50),
      child: InkWell(
        child: const Text(
          "¿Ya tienes cuenta? Inicia sesión",
          style: TextStyle(
              color: Colors.black87,
              decoration: TextDecoration.underline,
              fontWeight: FontWeight.w600
          ),
          textAlign: TextAlign.center,
        ),
        onTap: () {
          Navigator.push(context, MaterialPageRoute(
              builder: (context) => const SignIn()
          ));
        },
      ),
    );
  }
}

File: lib/src/presentation/widgets/SignInButton.java
package src.presentation.widgets;

import 'package:flutter/material.dart';
import 'package:tapp/core/themes/app_colors.dart';

class SignInButton extends StatelessWidget {
  final Function _signInFunction;

  const SignInButton(this._signInFunction);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 10),
      child: RaisedButton(
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(10),
        ),
        color: AppColors.purple,
        child: Text(
          'Entrar',
          style: TextStyle(
            fontWeight: FontWeight.w700,
          ),
        ),
        onPressed: _signInFunction,
      ),
    );
  }
}
